### PR TITLE
Add support for Choctopus44 keyboard

### PR DESCRIPTION
**Pull request by SlightHeadache**
## Description

Choctopus44 is a unibody split keyboard with fairly aggressive columnar stagger and a rotary encoder.

The default keymap is designed to be primarily used with Nordic languages, however I've not heard any complaints of the layout from international testers of the board.

While the PCB for the board will have some changes made for future revisions, from the firmware's point of view things are more or less set in stone.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/routa_keebs/choctopus44/config.h [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/config.h:165]: Lines 108-165 should be deleted.

```suggestion
```

File: keyboards/routa_keebs/choctopus44/config.h [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/config.h:198]: Same for Lines 179-198.

```suggestion
```

File: keyboards/routa_keebs/choctopus44/info.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                {"label": "K63", "x": 3.5, "y": 3.75},
+                {"label": "K64", "x": 4.5, "y": 4.0},
+                {"label": "K65", "x": 5.5, "y": 4.5},
+                
+                {"label": "K70", "x": 7, "y": 4.5},
+                {"label": "K71", "x": 8, "y": 4.0},
+                {"label": "K72", "x": 9, "y": 3.75},
+                {"label": "K73", "x": 10, "y": 3.75},
+                
+                {"label": "K60", "x": 6.25, "y": 3.25}
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/info.json:62]: `info.json` needs to have these objects in the same order that `keymap.c` uses.

```suggestion
                {"label": "K00", "x": 0, "y": 1.0},
                {"label": "K01", "x": 1, "y": 1.0},
                {"label": "K02", "x": 2, "y": 0.4},
                {"label": "K03", "x": 3, "y": 0},
                {"label": "K04", "x": 4, "y": 0.4},
                {"label": "K05", "x": 5, "y": 0.7},

                {"label": "K30", "x": 7.5, "y": 0.7},
                {"label": "K31", "x": 8.5, "y": 0.4},
                {"label": "K32", "x": 9.5, "y": 0},
                {"label": "K33", "x": 10.5, "y": 0.4},
                {"label": "K34", "x": 11.5, "y": 1.0},
                {"label": "K35", "x": 12.5, "y": 1.0},

                {"label": "K10", "x": 0, "y": 2.0},
                {"label": "K11", "x": 1, "y": 2.0},
                {"label": "K12", "x": 2, "y": 1.4},
                {"label": "K13", "x": 3, "y": 1.0},
                {"label": "K14", "x": 4, "y": 1.4},
                {"label": "K15", "x": 5, "y": 1.7},

                {"label": "K40", "x": 7.5, "y": 1.7},
                {"label": "K41", "x": 8.5, "y": 1.4},
                {"label": "K42", "x": 9.5, "y": 1.0},
                {"label": "K43", "x": 10.5, "y": 1.4},
                {"label": "K44", "x": 11.5, "y": 2.0},
                {"label": "K45", "x": 12.5, "y": 2.0},

                {"label": "K20", "x": 0, "y": 3.0},
                {"label": "K21", "x": 1, "y": 3.0},
                {"label": "K22", "x": 2, "y": 2.4},
                {"label": "K23", "x": 3, "y": 2.0},
                {"label": "K24", "x": 4, "y": 2.4},
                {"label": "K25", "x": 5, "y": 2.7},

                {"label": "K60", "x": 6.25, "y": 3.25},

                {"label": "K50", "x": 7.5, "y": 2.7},
                {"label": "K51", "x": 8.5, "y": 2.4},
                {"label": "K52", "x": 9.5, "y": 2.0},
                {"label": "K53", "x": 10.5, "y": 2.4},
                {"label": "K54", "x": 11.5, "y": 3.0},
                {"label": "K55", "x": 12.5, "y": 3.0},

                {"label": "K62", "x": 2.5, "y": 3.75},
                {"label": "K63", "x": 3.5, "y": 3.75},
                {"label": "K64", "x": 4.5, "y": 4.0},
                {"label": "K65", "x": 5.5, "y": 4.5},

                {"label": "K70", "x": 7, "y": 4.5},
                {"label": "K71", "x": 8, "y": 4.0},
                {"label": "K72", "x": 9, "y": 3.75},
                {"label": "K73", "x": 10, "y": 3.75}
```

File: keyboards/routa_keebs/choctopus44/info.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
+{
+    "keyboard_name": "Choctopus44",
+    "url": "",
+    "maintainer": "SlightHeadache",
+    "width": 13.5,
+    "height": 5.5,
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/info.json:6]:
```suggestion
```

File: keyboards/routa_keebs/choctopus44/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                        _______,  _______,  _______,   _______,            _______,  _______,  _______,  _______ \
+  ),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    switch(biton32(layer_state)) {
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/keymaps/default/keymap.c:63]:
```suggestion
    switch (get_highest_layer(state)) {
```

File: keyboards/routa_keebs/choctopus44/rules.mk [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/rules.mk:11]:
```suggestion
```

File: keyboards/routa_keebs/choctopus44/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+ENCODER_ENABLE = yes	    # Enable rotary encoder
```
Comment (noroadsleft) [inline keyboards/routa_keebs/choctopus44/rules.mk:32]: Formatting clean-up; no logic changes here.

```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes        # Enable rotary encoder
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/15398/files).

File: keyboards/routa_keebs/choctopus44/config.h [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* VID: 0xfeed as is used in many other QMK boards */
+/* PID: USB Device descriptor parameter, as per https://github.com/obdev/v-usb/blob/master/usbdrv/USB-IDs-for-free.txt */
+#define VENDOR_ID 0xfeed
+#define PRODUCT_ID 0x05dc
+#define DEVICE_VER 0x0001
+#define MANUFACTURER "routa.keebs@gmail.com"
```
Comment (drashna) [inline keyboards/routa_keebs/choctopus44/config.h:27]:
```suggestion
#define MANUFACTURER Routa Keebs
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**